Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM python:3.8.6-slim-buster as development
+
+WORKDIR /app
+
+ENV FLASK_RUN_HOST 0.0.0.0
+ENV FLASK_APP todo_app/app.py
+ENV FLASK_ENV development
+
+RUN pip install poetry
+RUN poetry config virtualenvs.create false
+
+COPY poetry.lock pyproject.toml /app/
+
+RUN poetry install -n
+
+EXPOSE 5000
+CMD ["flask", "run"]
+
+
+FROM python:3.8-slim-buster as production
+
+RUN mkdir /app 
+WORKDIR /app
+
+ENV PYTHONPATH=/app 
+ENV PYTHONUNBUFFERED True
+ENV PYTHONHASHSEED 0
+
+RUN pip install poetry
+RUN poetry config virtualenvs.create false
+
+COPY poetry.lock pyproject.toml /app/
```

Comment:
To avoid some of the repeated commands here you could have created a shared `base` build stage.